packages/services/src/rewards.service.ts

```diff
@@ -421,6 +421,162 @@ export class RewardsService {
     return results;
   }
 
+  /**
+   * Get rewards report data for a competition
+   * @param competitionId The competition ID (UUID) to generate report for
+   * @returns Report data including competition info, rewards stats, and merkle root
+   */
+  public async getRewardsReportData(competitionId: string): Promise<{
```

Comment:
calculates report data. I'm going with js computation to avoid complex queries, don't think it's a big deal to do in the js side because we have ~7k rewards, which is a small value